src/rustdoc-json-types/lib.rs

```diff
@@ -256,7 +254,12 @@ pub enum ItemEnum {
 
     TypeAlias(TypeAlias),
     OpaqueTy(OpaqueTy),
-    Constant(Constant),
+    Constant {
+        #[serde(rename = "type")]
+        type_: Type,
+        #[serde(rename = "const")]
+        const_: Constant,
```

Comment:
This is fine for now, but once this lands, I'd like to look into using seperate type for `Item::Constant` and `Term::Constant`/`GenericArgKind::Constant`, so this can be flattened again. But this shoudn't block this PR.

Reply:
As `cargo-semver-checks` maintainer, I'd love it if it's possible to make all the proposed representation changes quickly so that the separated representation is not the one that makes it into the next beta in ~9 days time.

If it goes into beta and then becomes stable, then I'll have to support both the separated representation and the newer form, and `cargo-semver-checks` will have to compile one more crate than otherwise — both negatives on my end.

If it's not possible to do that, and in your opinion it's better for the overall Rust community if I draw the short straw here, that's okay — I can handle it. I just would like to ask that it be a considered, intentional decision and not just unfortunate happenstance.

Reply:
Further discussion should go to #125974